gym_electric_motor/envs/gym_eesm/cont_cc_eesm_env.py

```diff
@@ -0,0 +1,165 @@
+from gym_electric_motor.core import ElectricMotorEnvironment, ReferenceGenerator, RewardFunction, \
+    ElectricMotorVisualization
+from gym_electric_motor.physical_systems.physical_systems import ExternallyExcitedSynchronousMotorSystem
+from gym_electric_motor.visualization import MotorDashboard
+from gym_electric_motor.reference_generators import WienerProcessReferenceGenerator, MultipleReferenceGenerator
+from gym_electric_motor import physical_systems as ps
+from gym_electric_motor.reward_functions import WeightedSumOfErrors
+from gym_electric_motor.utils import initialize
+from gym_electric_motor.constraints import LimitConstraint, SquaredConstraint
+
+
+class ContCurrentControlExternallyExcitedSynchronousMotorEnv(ElectricMotorEnvironment):
+    """
+    Description:
+        Environment to simulate a continuous control set current controlled externally excited synchronous motor.
+
+    Key:
+        ``'Cont-CC-EESM-v0'``
+
+    Default Components:
+        - Supply: :py:class:`.IdealVoltageSupply`
+        - Converter: :py:class:`.ContB6BridgeConverter`
+        - Motor: :py:class:`.ExternallyExcitedSynchronousMotor`
+        - Load: :py:class:`.ConstantSpeedLoad`
+        - Ode-Solver: :py:class:`.ScipyOdeSolver`
+
+        - Reference Generator: :py:class:`.WienerProcessReferenceGenerator` *Reference Quantity:* ``'i_sd', 'i_sq``
+
+        - Reward Function: :py:class:`.WeightedSumOfErrors` reward_weights: ``'i_sd' = 0.5, 'i_sq' = 0.5``
+
+        - Visualization: :py:class:`.MotorDashboard` current and action plots
+
+        - Constraints: :py:class:`.SquaredConstraint` on the currents  ``'i_sd', 'i_sq'``
+
+    State Variables:
+        ``['omega' , 'torque', 'i_sd', 'i_sq', 'i_a', 'i_b', 'i_c', 'i_e', 'u_sd', 'u_sq', 'u_a', 'u_b', 'u_c', 'u_e', 'u_sup']``
+
+    Reference Variables:
+        ``['i_sd', 'i_sq']``
+
+    Control Cycle Time:
+        tau = 1e-4 seconds
+
+    Observation Space:
+        Type: Tuple(State_Space, Reference_Space)
+
+    State Space:
+        Box(low= 15 * [-1], high= 15 * [1])
+
+    Reference Space:
+        Box(low=[-1, -1, -1], high=[1, 1, 1])
```

Comment:
I assume this is for i_d, i_q and i_e instead of i_abc?
If yes, i_e would have Box constraints in the interval [0, 1]